src/scaladoc/scala/tools/nsc/doc/html/Doclet.scala

```diff
@@ -12,9 +12,17 @@ import doclet._
 /** The default doclet used by the scaladoc command line tool
   * when no user-provided doclet is provided. */
 class Doclet extends Generator with Universer {
+  import scala.reflect.internal.Reporter
 
-  def generateImpl() {
-    new html.HtmlFactory(universe, new ScalaDocReporter(universe.settings)).generate()
```

Comment:
the change here (and the type change from `ScalaDocReporter` to `Reporter` in various places`) seems to be unrelated to delayed reporting - what was the motivation?